jlc_lib/generate_bom.py

```diff
@@ -42,25 +42,59 @@ def GenerateBOM(input_filename, output_filename):
   num_groups_found = 0
   for group in grouped:
     refs = []
+    lcsc_part_numbers = []
+    footprints = []
+
     for component in group:
-        refs.append(component.getRef())
-        c = component
-
-    # Get the field name for the LCSC part number.
-    lcsc_part_number = ""
-    for field_name in c.getFieldNames():
-      field_value = c.getField(field_name)
-      if LCSC_PART_NUMBER_MATCHER.match(field_value):
-        lcsc_part_number = field_value
-
-    # Skip groups without LCSC part number.
-    if lcsc_part_number == "":
+      refs.append(component.getRef())
+      c = component
+
+      # Get the field name for the LCSC part number.
+      for field_name in c.getFieldNames():
+        field_value = c.getField(field_name)
+
+        if LCSC_PART_NUMBER_MATCHER.match(field_value):
+          lcsc_part_numbers.append(field_value)
+        else:
+          lcsc_part_numbers.append(None)
+
+      footprints.append(c.getFootprint())
+
+
+    # Check part numbers for uniqueness
+    lcsc_part_numbers_set = set(lcsc_part_numbers)
```

Comment:
Why not just build this set incrementally as we go through components? Also no need for lcsc_part_numbers_set_without_none in that case if we don't insert Nones.

Reply:
lcsc_part_numbers_set (with none) is used to check if part number was forgotten on some parts (see line 78).
The idea of using a list in the beginning was that I could exactly name the components which different/missing part numbers. But then decided on a shorter logging, so removed this now.